### PR TITLE
`MFX_Timing` - updated event sequences.

### PR DESCRIPTION
**Pull request by fredericpoitevin**
Event sequences for 20Hz and 30Hz updated to reflect what worked during mfxl1010021.

(We should make this code prettier eventually but I went for efficiency here)

### CODE REVIEW

File: mfx/macros.py [hunk 1]
```diff
@@ -88,50 +88,64 @@ def __init__(self,sequencer=None):
...
-        self.wait1 = [0,1,0,0]
-        self.zeros = [0,0,0,0]
-        self.laser_on = [203,0,0,0]
-        self.laser_off = [204,0,0,0]
+        self.evt_code_names = ['wait','pp_trig','daq_readout',
+                               'laser_on','laser_off',
+                               'ray_readout','ray1','ray2','ray3']
+        self.evt_code_ids = [0,197,198,
+                             203,204,
+                             210,211,212,213]
```
Comment (lg345) [inline mfx/macros.py:96]: We might swap this to a dictionary for readability some day.

File: mfx/macros.py [hunk 2]
```diff
@@ -88,50 +88,64 @@ def __init__(self,sequencer=None):
...
+        #self.pp_trig = [197,0,0,0]
+        #self.daq_readout = [198,0,0,0]
+        #self.wait1 = [0,1,0,0]
+        #self.zeros = [0,0,0,0]
+        #self.laser_on = [203,0,0,0]
+        #self.laser_off = [204,0,0,0]
         self.sync_markers = {0.5:0, 1:1, 5:2, 10:3, 30:4, 60:5, 120:6, 360:7}
+    def _step(self, evt_code_name=None, delta_beam=0):
+        try:
+            evt_code_id = self.evt_code_ids[self.evt_code_names.index(evt_code_name)]
```
Comment (lg345) [inline mfx/macros.py:110]: On second thought I'm nearly certain this would be simpler with a dictionary.

File: mfx/macros.py [hunk 3]
```diff
@@ -88,50 +88,64 @@ def __init__(self,sequencer=None):
...
+            return [evt_code_id, delta_beam, 0, 0]
```
Comment (lg345) [inline mfx/macros.py:111]: If we give the sequencer a fifth element that is the string describing the event code, will it populate that description box in the event sequencer? Would be nice for readability in the event sequencer. I would test it but all of the filesystems are down at the LCLS.

File: mfx/macros.py (hunk 2 above)
Comment (fredericpoitevin) [reply mfx/macros.py:110]: Yes good idea, will do!

File: mfx/macros.py (hunk 3 above)
Comment (fredericpoitevin) [reply mfx/macros.py:111]: OK that'd be nice indeed - let's report here when one of us has been able to test

Comment (fredericpoitevin) [reply mfx/macros.py:111]: Just checked and unfortunately there doesn't seem to be a method to do that.

The sequence object does not have a field for it:
```python
In [1]: from mfx.db import sequencer as seq

In [2]: seq.sequence
Out[2]:

mfx_sequencer_sequence
----------------------
units: None
ec_array: [213 212 210 ...   0   0   0]
bd_array: [1 1 1 ... 0 0 0]
fd_array: [0 0 0 ... 0 0 0]
bc_array: [0 0 0 ... 0 0 0]
seq_proc: 1

In [3]: seq.get()
Out[3]: EventSequencerTuple(play_control=0, sequence_length=8, current_step=0, play_count=0, total_play_count=0, play_status=0, play_mode=2, sync_marker=5, next_sync=0, pulse_req=0, rep_count=500, sequence_owner=7, sequence=EventSequenceTuple(ec_array=array([213, 212, 210, ...,   0,   0,   0], dtype=int32), bd_array=array([1, 1, 1, ..., 0, 0, 0], dtype=int32), fd_array=array([0, 0, 0, ..., 0, 0, 0], dtype=int32), bc_array=array([0, 0, 0, ..., 0, 0, 0], dtype=int32), seq_proc=1))
```

I've tried anyway and it did not work:
```python
In [4]: sequence = seq.sequence.get_seq()
In [4]: sequence.append([0,0,0,0,'test'])
In [5]: sequence
Out[5]:
[[213, 1, 0, 0],
 [212, 1, 0, 0],
 [210, 1, 0, 0],
 [197, 1, 0, 0],
 [212, 0, 0, 0],
 [211, 1, 0, 0],
 [210, 1, 0, 0],
 [198, 0, 0, 0],
 [0, 0, 0, 0, 'test']]
In [6]: seq.sequence.put_seq(sequence)
```
The step was added, but without the text:
image

Comment (lg345) [reply mfx/macros.py]: Damn. We should write our own caputter to fill the device/function (just kidding. don't do this)

Comment (fredericpoitevin) [reply mfx/macros.py]: Yeah it's tempting... maybe for HE ;)

**Review by lg345 — APPROVED** [review]
Great work. It looks like it will work programatically. Hopefully it will work experimentally.

**Pull request merged**